### ISSUE TITLE
Bug: [@typescript-eslint/no-inferrable-types] ignores readonly properties

### ISSUE THREAD

**Issue by Ionaru** [typescript-eslint/typescript-eslint#10456]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=MYGwhgzhAEBiD29oG8CwAoa0BOBTMAJvAHYgCe0ADtvJQFzTECuAtgEa7bQC80ArAG4MAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDNFpoAQwBGSWkVKoMkftA7RI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Foo {
  readonly prop: number = 5;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-inferrable-types": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected that the 2nd line should report: "Type number trivially inferred from a number literal, remove type annotation."

### Actual Result

The linter passes without errors.

### Additional Info

Removing `readonly` will correctly trigger the error.

**Comment by ronami** [comment]
I think it's correct to ignore `readonly` properties, as adding the type annotation changes the type of the expression (it's `number` with a type annotation and `5` without it).

I think this was taken into consideration when the rule was written: https://github.com/typescript-eslint/typescript-eslint/blob/c92c29d142640b107795cfcdd28e94e4a811872d/packages/eslint-plugin/src/rules/no-inferrable-types.ts#L270-L273

**Comment by Ionaru** [comment]
While I see the issue, I don't think it's correct to completely ignore a `readonly` property. Especially when it is never reassigned. As far as I can see, there's no reason to widen the type of a `readonly` property when there is no constructor to change it.

`readonly prop: 5 = 5;` is also allowed under this rule's `readonly` exception. (but it trips `@typescript-eslint/ prefer-as-const`)

Somehow, `readonly prop: 5 = 5 as const;` is valid again, as is `readonly prop: number = 5 as const;`. Both still allow reassignment in the constructor.

**Comment by bradzacher** [comment]
[playground example showing why this is ignored](https://www.typescriptlang.org/play/#src=export%20class%20Hello%20%7B%0D%0A%20%20%20%20readonly%20foo%20%3D%20false%3B%0D%0A%20%20%20%20constructor(foo%3A%20boolean%7Cundefined)%20%7B%0D%0A%20%20%20%20%20%20%20%20this.foo%20%3D%20foo%20%7C%7C%20this.foo%3B%0D%0A%20%20%20%20%7D%0D%0A%7D)

Ignoring this is entirely intentional because of how TS derives the type of the property.

> I don't think it's correct to completely ignore a readonly property. Especially when it is never reassigned.

Most people are okay with ignoring this always because it means they can just annotate all readonly properties and then never have to worry about this class of TS error.

[This behaviour has existed in the rule since the rule's creation over 6y ago.](https://github.com/bradzacher/eslint-plugin-typescript/pull/124/files#diff-199cdcaf055348c6709332bc23b233d41e96213a9739b170953044d2063e10a7R173). This is the first time someone has asked for the behaviour to change.

So this is working as intended and it's the behaviour most users seem to expect.

For these reasons I shall close this issue.
